integration test flake: TestUpdateSelectorToRemoveControllerRef

https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/29205/kubernetes-pull-test-unit-integration/37505/

```
--- FAIL: TestUpdateSelectorToRemoveControllerRef (70.13s)
	replicaset_test.go:367: patched rs = &v1beta1.ReplicaSet{TypeMeta:unversioned.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:v1.ObjectMeta{Name:"rs", GenerateName:"", Namespace:"rs-update-selector-to-remove-controllerref", SelfLink:"/apis/extensions/v1beta1/namespaces/rs-update-selector-to-remove-controllerref/replicasets/rs", UID:"9e23e3d7-5979-11e6-88da-0242ac110002", ResourceVersion:"3097", Generation:2, CreationTimestamp:unversioned.Time{Time:time.Time{sec:63605825756, nsec:0, loc:(*time.Location)(0x2f06c80)}}, DeletionTimestamp:(*unversioned.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string{"name":"test"}, Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference(nil), Finalizers:[]string(nil)}, Spec:v1beta1.ReplicaSetSpec{Replicas:(*int32)(0xc820e3aea8), Selector:(*v1beta1.LabelSelector)(0xc8212ea3e0), Template:v1.PodTemplateSpec{ObjectMeta:v1.ObjectMeta{Name:"", GenerateName:"", Namespace:"", SelfLink:"", UID:"", ResourceVersion:"", Generation:0, CreationTimestamp:unversioned.Time{Time:time.Time{sec:0, nsec:0, loc:(*time.Location)(nil)}}, DeletionTimestamp:(*unversioned.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string{"name":"test", "uniqueKey":"1"}, Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference(nil), Finalizers:[]string(nil)}, Spec:v1.PodSpec{Volumes:[]v1.Volume(nil), InitContainers:[]v1.Container(nil), Containers:[]v1.Container{v1.Container{Name:"fake-name", Image:"fakeimage", Command:[]string(nil), Args:[]string(nil), WorkingDir:"", Ports:[]v1.ContainerPort(nil), Env:[]v1.EnvVar(nil), Resources:v1.ResourceRequirements{Limits:v1.ResourceList(nil), Requests:v1.ResourceList(nil)}, VolumeMounts:[]v1.VolumeMount(nil), LivenessProbe:(*v1.Probe)(nil), ReadinessProbe:(*v1.Probe)(nil), Lifecycle:(*v1.Lifecycle)(nil), TerminationMessagePath:"/dev/termination-log", ImagePullPolicy:"Always", SecurityContext:(*v1.SecurityContext)(nil), Stdin:false, StdinOnce:false, TTY:false}}, RestartPolicy:"Always", TerminationGracePeriodSeconds:(*int64)(0xc820e3b118), ActiveDeadlineSeconds:(*int64)(nil), DNSPolicy:"ClusterFirst", NodeSelector:map[string]string(nil), ServiceAccountName:"", DeprecatedServiceAccount:"", NodeName:"", HostNetwork:false, HostPID:false, HostIPC:false, SecurityContext:(*v1.PodSecurityContext)(0xc820f81f00), ImagePullSecrets:[]v1.LocalObjectReference(nil), Hostname:"", Subdomain:""}}}, Status:v1beta1.ReplicaSetStatus{Replicas:2, FullyLabeledReplicas:2, ObservedGeneration:1}}
	replicaset_test.go:117: expect 3 pods, got 4 pods
	replicaset_test.go:117: expect 3 pods, got 4 pods
	replicaset_test.go:117: expect 3 pods, got 4 pods
	replicaset_test.go:117: expect 3 pods, got 4 pods
	replicaset_test.go:117: expect 3 pods, got 4 pods
	replicaset_test.go:117: expect 3 pods, got 4 pods
	replicaset_test.go:117: expect 3 pods, got 4 pods
	replicaset_test.go:372: timed out waiting for the condition
```

--- a/test/integration/replicaset/replicaset_test.go
+++ b/test/integration/replicaset/replicaset_test.go
@@ -316,7 +316,6 @@ func TestUpdateSelectorToAdopt(t *testing.T) {
 
 	stopCh := make(chan struct{})
 	go podInformer.Run(stopCh)
-	waitToObservePods(t, podInformer, 2)
 	go rm.Run(5, stopCh)
 	waitRSStable(t, clientSet, rs, ns.Name)
 
@@ -354,6 +353,7 @@ func TestUpdateSelectorToRemoveControllerRef(t *testing.T) {
 
 	stopCh := make(chan struct{})
 	go podInformer.Run(stopCh)
+	waitToObservePods(t, podInformer, 2)
 	go rm.Run(5, stopCh)
 	waitRSStable(t, clientSet, rs, ns.Name)
 

